[Automation] native-image run fails for org.apache.avro:avro-ipc:1.12.0

## native-image run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.12.0" ./gradlew clean test -Pcoordinates="org.apache.avro:avro-ipc:1.10.2"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25594561246/job/75138264409
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
h@driver=/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.avro/avro-ipc/1.10.2/build/native/nativeTestCompile \
-H:BuildOutputJSONFile@user=/tmp/native-image-build-output.json \
-H:ReachabilityMetadataResources@file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.avro/avro-ipc/1.10.2/build/resources/test/META-INF/native-image/reachability-metadata.json+api=META-INF/native-image/reachability-metadata.json \
-H:FallbackThreshold@user+api=0 \
-H:Name@user+api=org.apache.avro.avro-ipc_tests-tests \
-H:ConfigurationFileDirectories@user=/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.avro/avro-ipc/1.10.2/build/native/generated/generateTestResourcesConfigFile,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.opentest4j/opentest4j/1.2.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-commons/1.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-params/5.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-engine/5.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-engine/1.10.4,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-launcher/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-reporting/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.vintage/junit-vintage-engine/5.9.1,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/junit/junit/4.13.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.hamcrest/hamcrest-core/1.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.apache.avro/avro-ipc/1.10.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.fasterxml.jackson.core/jackson-core/2.6.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.fasterxml.jackson.core/jackson-annotations/2.3.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.fasterxml.jackson.core/jackson-databind/2.15.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.apache.commons/commons-compress/1.23.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/commons-codec/commons-codec/1.10,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/commons-io/commons-io/2.11.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.apache.commons/commons-lang3/3.11,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.slf4j/slf4j-api/1.7.36,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.apache.velocity/velocity-engine-core/2.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/javax.annotation/javax.annotation-api/1.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.tukaani/xz/1.8,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.github.luben/zstd-jni/1.5.7-2 \
-H:Optimize@user+api=b \
-H:+ReportExceptionStackTraces@user \
-H:Features@user+api=org.graalvm.junit.platform.JUnitPlatformFeature \
'-H:Class@explicit main-class=org.graalvm.junit.platform.NativeImageJUnitLauncher' \
-H:ImageBuildID@driver=e0a13d47-600d-b1e3-4407-8fc563f805b5 \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature,com.oracle.svm.polyglot.scala.ScalaFeature'
]
========================================================================================================================
GraalVM Native Image: Generating 'org.apache.avro.avro-ipc_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                    (6.3s @ 0.16GB)
 Java version: 25.0.3+9-LTS, vendor version: Oracle GraalVM 25.0.3+9.1
 Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 C compiler: gcc (linux, x86_64, 13.3.0)
 Garbage collector: Serial GC (max heap size: 80% of RAM)
 2 user-specific feature(s):
 - com.oracle.svm.thirdparty.gson.GsonFeature
 - org.graalvm.junit.platform.JUnitPlatformFeature
------------------------------------------------------------------------------------------------------------------------
 1 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 12.67GB of memory (75.6% of system memory, $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.avro/avro-ipc/1.10.2/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                   (37.9s @ 1.02GB)
    9,518 types,  11,133 fields, and  43,122 methods found reachable
    3,315 types,   1,161 fields, and   4,835 methods registered for reflection
       68 types,      78 fields, and      52 methods registered for JNI access
        0 downcalls and 0 upcalls registered for foreign access
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                               (5.6s @ 0.97GB)
[4/8] Parsing methods...      [**]                                                                       (3.9s @ 1.02GB)
[5/8] Inlining methods...     [***]                                                                      (2.3s @ 1.09GB)
[6/8] Compiling methods...    [*****]                                                                   (21.1s @ 0.81GB)
[7/8] Laying out methods...   [**]                                                                       (3.1s @ 1.01GB)
[8/8] Creating image...       [**]                                                                       (3.6s @ 1.17GB)
  11.43MB (24.76%) for code area:    26,950 compilation units
  30.41MB (65.86%) for image heap:  234,900 objects and 174 resources
   4.33MB ( 9.38%) for other data
  46.17MB in total image size, 43.78MB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   4.68MB java.base                                           16.18MB byte[] for general heap data
   2.39MB svm.jar (Native Image)                               3.35MB byte[] for code metadata
 943.43kB jackson-databind-2.17.2.jar                          2.95MB byte[] for java.lang.String
 554.15kB assertj-core-3.22.0.jar                              1.55MB java.lang.String
 534.86kB java.xml                                             1.20MB java.lang.Class
 370.59kB jackson-core-2.17.2.jar                            989.87kB com.oracle.svm.core.hub.DynamicHubCompanion
 326.15kB avro-1.12.0.jar                                    453.66kB java.util.concurrent.ConcurrentHashMap$Node
 198.48kB junit-jupiter-engine-5.9.2.jar                     439.44kB byte[] for reflection metadata
 121.76kB junit-platform-launcher-1.9.2.jar                  305.86kB java.util.HashMap$Node
 103.17kB java.logging                                       288.31kB java.lang.Object[]
 789.82kB for 33 more packages                                 2.69MB for 2195 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 28 component(s) is embedded in binary (1.34kB). 9 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 FUTR: Use '--future-defaults=all' to prepare for future releases.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                        6.3s (7.3% of total time) in 853 GCs | Peak RSS: 2.01GB | CPU load: 3.63
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.avro/avro-ipc/1.10.2/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.avro/avro-ipc/1.10.2/build/native/nativeTestCompile/org.apache.avro.avro-ipc_tests-tests (executable)
 /tmp/native-image-build-output.json (build_info)
========================================================================================================================
Finished generating 'org.apache.avro.avro-ipc_tests-tests' in 1m 26s.
[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.avro/avro-ipc/1.10.2/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.avro/avro-ipc/1.10.2/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.avro/avro-ipc/1.10.2/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

org_apache_avro.avro_ipc.ReflectRequestorTest > getClientWithExistingReflectRequestorCreatesProxyUsingRequestorClassLoader() FAILED

org_apache_avro.avro_ipc.ReflectRequestorTest > getClientWithReflectDataCreatesProxyThatReportsProtocolAndRemoteName() FAILED

org_apache_avro.avro_ipc.SpecificRequestorTest > getClientCreatesProxyThatReportsProtocolAndRemoteName() SUCCESSFUL

org_apache_avro.avro_ipc.SpecificRequestorTest > getClientWithExistingRequestorCreatesProxyUsingRequestorClassLoader() SUCCESSFUL

org_apache_avro.avro_ipc.SpecificResponderTest > respondInvokesMatchingImplementationMethod() SUCCESSFUL


Failures (2):
  JUnit Jupiter:ReflectRequestorTest:getClientWithExistingReflectRequestorCreatesProxyUsingRequestorClassLoader()
    MethodSource [className = 'org_apache_avro.avro_ipc.ReflectRequestorTest', methodName = 'getClientWithExistingReflectRequestorCreatesProxyUsingRequestorClassLoader', methodParameterTypes = '']
    => java.lang.ExceptionInInitializerError
       org.apache.avro.reflect.ReflectData.getProtocol(ReflectData.java:903)
       org.apache.avro.ipc.reflect.ReflectRequestor.<init>(ReflectRequestor.java:46)
       org_apache_avro.avro_ipc.ReflectRequestorTest.getClientWithExistingReflectRequestorCreatesProxyUsingRequestorClassLoader(ReflectRequestorTest.java:40)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
     Caused by: org.apache.avro.AvroRuntimeException: Unable to load a functional FieldAccess class!
       org.apache.avro.reflect.ReflectionUtil.resetFieldAccess(ReflectionUtil.java:64)
       org.apache.avro.reflect.ReflectionUtil.<clinit>(ReflectionUtil.java:51)
       [...]
  JUnit Jupiter:ReflectRequestorTest:getClientWithReflectDataCreatesProxyThatReportsProtocolAndRemoteName()
    MethodSource [className = 'org_apache_avro.avro_ipc.ReflectRequestorTest', methodName = 'getClientWithReflectDataCreatesProxyThatReportsProtocolAndRemoteName', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class org.apache.avro.reflect.ReflectionUtil
       org.apache.avro.reflect.ReflectData.getProtocol(ReflectData.java:903)
       org.apache.avro.ipc.reflect.ReflectRequestor.getClient(ReflectRequestor.java:75)
       org_apache_avro.avro_ipc.ReflectRequestorTest.getClientWithReflectDataCreatesProxyThatReportsProtocolAndRemoteName(ReflectRequestorTest.java:26)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
       java.base@25.0.3/java.util.concurrent.FutureTask.run(FutureTask.java:328)
       java.base@25.0.3/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
       [...]

Test run finished after 7 ms
[         5 containers found      ]
[         0 containers skipped    ]
[         5 containers started    ]
[         0 containers aborted    ]
[         5 containers successful ]
[         0 containers failed     ]
[         5 tests found           ]
[         0 tests skipped         ]
[         5 tests started         ]
[         0 tests aborted         ]
[         3 tests successful      ]
[         2 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.avro/avro-ipc/1.10.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
18 actionable tasks: 8 executed, 10 up-to-date
SLF4J(W): No SLF4J providers were found.
SLF4J(W): Defaulting to no-operation (NOP) logger implementation
SLF4J(W): See https://www.slf4j.org/codes.html#noProviders for further details.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.avro/avro-ipc/1.10.2/build/native/nativeTestCompile/org.apache.avro.avro-ipc_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1m 29s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.apache.avro:avro-ipc:1.10.2 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 31s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 1.12.0 at 2026-05-09T07:10:06Z with exit code 1 after 91s
❌ ERROR: 'native-image run' for 1.12.0 failed with exit code 1.
FAILED[native-image run][1.12.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.12.0" ./gradlew clean test -Pcoordinates="org.apache.avro:avro-ipc:1.10.2"]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.12.0" ./gradlew clean test -Pcoordinates="org.apache.avro:avro-ipc:1.10.2"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25594561246/job/75138264382
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
.10.2/build/native/generated/generateTestResourcesConfigFile,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.opentest4j/opentest4j/1.2.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-commons/1.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-params/5.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-engine/5.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-engine/1.10.4,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-reporting/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-launcher/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.vintage/junit-vintage-engine/5.9.1,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/junit/junit/4.13.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.hamcrest/hamcrest-core/1.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.apache.avro/avro-ipc/1.10.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.fasterxml.jackson.core/jackson-core/2.6.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.fasterxml.jackson.core/jackson-annotations/2.3.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.fasterxml.jackson.core/jackson-databind/2.15.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.apache.commons/commons-compress/1.23.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/commons-codec/commons-codec/1.10,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/commons-io/commons-io/2.11.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.apache.commons/commons-lang3/3.11,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.slf4j/slf4j-api/1.7.36,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.apache.velocity/velocity-engine-core/2.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/javax.annotation/javax.annotation-api/1.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.tukaani/xz/1.8,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.github.luben/zstd-jni/1.5.7-2 \
-H:Optimize@user+api=b \
-H:+ReportExceptionStackTraces@user \
-H:Features@user+api=org.graalvm.junit.platform.JUnitPlatformFeature \
'-H:Class@explicit main-class=org.graalvm.junit.platform.NativeImageJUnitLauncher' \
-H:ImageBuildID@driver=9c11e7c5-dc7c-5a3e-837b-036fa04d75e0 \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature' \
'-H:AddExports@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=java.base/jdk.internal.misc=org.graalvm.nativeimage.guest' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=com.oracle.svm.guest.features.jdk.hosted.JavaNetHttpFeature' \
'-H:ClassInitialization@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=sun.net.httpserver.simpleserver:run_time' \
'-H:ReachabilityMetadataResources@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json+api=META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json' \
-H:DriverWarningsCount=3
]
Warning: Option 'FallbackThreshold' is deprecated and might be removed in a future release: It no longer has any effect, and no replacement is available. Please refer to the GraalVM release notes.
========================================================================================================================
GraalVM Native Image: Generating 'org.apache.avro.avro-ipc_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                   (7.2s @ 0.23GiB)
 Builder configuration:
 - Java version: 25.0.2+10-LTS, vendor version: Oracle GraalVM 25.1.0-dev+10.1
 - Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 - C compiler: gcc (linux, x86_64, 13.3.0)
 - Assertions: disabled, system assertions: disabled
 - 1 user-specific feature(s):
   - org.graalvm.junit.platform.JUnitPlatformFeature
 Image configuration:
 - Garbage collector: Serial GC (max heap size: 80% of RAM)
 - Assertions: disabled (class-specific config may apply), system assertions: disabled
------------------------------------------------------------------------------------------------------------------------
 1 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 13.27GiB of memory (85% of system memory because $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.avro/avro-ipc/1.10.2/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                  (37.4s @ 0.94GiB)
    9,456 types,  11,034 fields, and  43,283 methods found reachable
      382 types,     119 fields, and   1,049 methods registered for reflection
       60 types,      68 fields, and      52 methods registered for JNI access
      155 resource accesses registered with 262.47KiB total size
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                              (4.7s @ 1.08GiB)
[4/8] Parsing methods...      [**]                                                                      (4.0s @ 0.72GiB)
[5/8] Inlining methods...     [***]                                                                     (2.1s @ 0.77GiB)
[6/8] Compiling methods...    [*****]                                                                  (21.6s @ 0.77GiB)
[7/8] Laying out methods...   [**]                                                                      (3.2s @ 0.97GiB)
[8/8] Creating image...       [**]                                                                      (3.6s @ 1.13GiB)
   9.98MiB (34.59%) for code area:    27,097 compilation units
  14.69MiB (50.89%) for image heap:  236,455 objects and 23 resources
   4.19MiB (14.52%) for other data
  28.86MiB in total image size, 26.56MiB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   1.09MiB o.g.n.runtime/com.oracle.svm.core.code              3.13MiB byte[] for code metadata
 741.05KiB java.base/java.util                                 2.84MiB byte[] for string data
 408.48KiB java.base/jdk.internal.classfile.impl               1.51MiB java.base/java.lang.String
 336.01KiB java.base/java.lang                                 1.14MiB java.base/java.lang.Class
 276.16KiB java.base/java.text                                 1.01MiB o.g.n.runtime/c.o.s.core.hub.DynamicHubCompanion
 246.51KiB java.base/java.util.concurrent                    593.52KiB byte[] for general heap data
 230.88KiB com.fasterxml.jackson.databind.deser.std          490.38KiB byte[] for reflection metadata
 224.55KiB java.base/com.sun.crypto.provider                 419.98KiB heap alignment
 223.43KiB org.assertj.core.internal.bytebuddy.jar.asm       365.70KiB o.g.n.runtime/c.o.s.c.h.ImageReflectionMetadata
 162.88KiB java.base/java.lang.invoke                        304.91KiB java.base/java.util.HashMap$Node
   6.11MiB for 359 more packages                               2.93MiB for 2203 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 23 component(s) is embedded in binary (1.21KiB). 9 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 FUTR: Use '--future-defaults=all' to prepare for future releases.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                       5.8s (6.6% of total time) in 832 GCs | Peak RSS: 1.78GiB | CPU load: 3.61
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.avro/avro-ipc/1.10.2/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.avro/avro-ipc/1.10.2/build/native/nativeTestCompile/org.apache.avro.avro-ipc_tests-tests (executable, 26.56MiB)
 /tmp/native-image-build-output.json (build_info, 1.04KiB)
========================================================================================================================
Finished generating 'org.apache.avro.avro-ipc_tests-tests' in 1m 27s.

The build process encountered 4 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.avro/avro-ipc/1.10.2/build/native/nativeTestCompile
SLF4J(W): No SLF4J providers were found.
SLF4J(W): Defaulting to no-operation (NOP) logger implementation
SLF4J(W): See https://www.slf4j.org/codes.html#noProviders for further details.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.avro/avro-ipc/1.10.2/build/native/nativeTestCompile/org.apache.avro.avro-ipc_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1m 31s

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.avro/avro-ipc/1.10.2/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.avro/avro-ipc/1.10.2/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

org_apache_avro.avro_ipc.ReflectRequestorTest > getClientWithExistingReflectRequestorCreatesProxyUsingRequestorClassLoader() FAILED

org_apache_avro.avro_ipc.ReflectRequestorTest > getClientWithReflectDataCreatesProxyThatReportsProtocolAndRemoteName() FAILED

org_apache_avro.avro_ipc.SpecificRequestorTest > getClientCreatesProxyThatReportsProtocolAndRemoteName() SUCCESSFUL

org_apache_avro.avro_ipc.SpecificRequestorTest > getClientWithExistingRequestorCreatesProxyUsingRequestorClassLoader() SUCCESSFUL

org_apache_avro.avro_ipc.SpecificResponderTest > respondInvokesMatchingImplementationMethod() SUCCESSFUL


Failures (2):
  JUnit Jupiter:ReflectRequestorTest:getClientWithExistingReflectRequestorCreatesProxyUsingRequestorClassLoader()
    MethodSource [className = 'org_apache_avro.avro_ipc.ReflectRequestorTest', methodName = 'getClientWithExistingReflectRequestorCreatesProxyUsingRequestorClassLoader', methodParameterTypes = '']
    => java.lang.ExceptionInInitializerError
       org.apache.avro.reflect.ReflectData.getProtocol(ReflectData.java:903)
       org.apache.avro.ipc.reflect.ReflectRequestor.<init>(ReflectRequestor.java:46)
       org_apache_avro.avro_ipc.ReflectRequestorTest.getClientWithExistingReflectRequestorCreatesProxyUsingRequestorClassLoader(ReflectRequestorTest.java:40)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
     Caused by: org.apache.avro.AvroRuntimeException: Unable to load a functional FieldAccess class!
       org.apache.avro.reflect.ReflectionUtil.resetFieldAccess(ReflectionUtil.java:64)
       org.apache.avro.reflect.ReflectionUtil.<clinit>(ReflectionUtil.java:51)
       [...]
  JUnit Jupiter:ReflectRequestorTest:getClientWithReflectDataCreatesProxyThatReportsProtocolAndRemoteName()
    MethodSource [className = 'org_apache_avro.avro_ipc.ReflectRequestorTest', methodName = 'getClientWithReflectDataCreatesProxyThatReportsProtocolAndRemoteName', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class org.apache.avro.reflect.ReflectionUtil
       org.apache.avro.reflect.ReflectData.getProtocol(ReflectData.java:903)
       org.apache.avro.ipc.reflect.ReflectRequestor.getClient(ReflectRequestor.java:75)
       org_apache_avro.avro_ipc.ReflectRequestorTest.getClientWithReflectDataCreatesProxyThatReportsProtocolAndRemoteName(ReflectRequestorTest.java:26)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
       java.base@25.0.2/java.util.concurrent.FutureTask.run(FutureTask.java:328)
       java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
       [...]

Test run finished after 7 ms
[         5 containers found      ]
[         0 containers skipped    ]
[         5 containers started    ]
[         0 containers aborted    ]
[         5 containers successful ]
[         0 containers failed     ]
[         5 tests found           ]
[         0 tests skipped         ]
[         5 tests started         ]
[         0 tests aborted         ]
[         3 tests successful      ]
[         2 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.avro/avro-ipc/1.10.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
18 actionable tasks: 8 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.apache.avro:avro-ipc:1.10.2 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 32s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 1.12.0 at 2026-05-09T07:10:54Z with exit code 1 after 93s
❌ ERROR: 'native-image run' for 1.12.0 failed with exit code 1.
FAILED[native-image run][1.12.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.12.0" ./gradlew clean test -Pcoordinates="org.apache.avro:avro-ipc:1.10.2"]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="1.12.0" ./gradlew clean test -Pcoordinates="org.apache.avro:avro-ipc:1.10.2"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25594561246/job/75138264412
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
.10.2/build/native/generated/generateTestResourcesConfigFile,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.opentest4j/opentest4j/1.2.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-commons/1.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-params/5.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-engine/5.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-engine/1.10.4,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-reporting/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-launcher/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.vintage/junit-vintage-engine/5.9.1,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/junit/junit/4.13.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.hamcrest/hamcrest-core/1.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.apache.avro/avro-ipc/1.10.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.fasterxml.jackson.core/jackson-core/2.6.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.fasterxml.jackson.core/jackson-annotations/2.3.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.fasterxml.jackson.core/jackson-databind/2.15.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.apache.commons/commons-compress/1.23.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/commons-codec/commons-codec/1.10,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/commons-io/commons-io/2.11.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.apache.commons/commons-lang3/3.11,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.slf4j/slf4j-api/1.7.36,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.apache.velocity/velocity-engine-core/2.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/javax.annotation/javax.annotation-api/1.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.tukaani/xz/1.8,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.github.luben/zstd-jni/1.5.7-2 \
-H:Optimize@user+api=b \
-H:+ReportExceptionStackTraces@user \
-H:FutureDefaults@user+api=all \
-H:Features@user+api=org.graalvm.junit.platform.JUnitPlatformFeature \
'-H:Class@explicit main-class=org.graalvm.junit.platform.NativeImageJUnitLauncher' \
-H:ImageBuildID@driver=97fc23b6-f77b-961c-be30-e7df827c8832 \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature' \
'-H:AddExports@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=java.base/jdk.internal.misc=org.graalvm.nativeimage.guest' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=com.oracle.svm.guest.features.jdk.hosted.JavaNetHttpFeature' \
'-H:ClassInitialization@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=sun.net.httpserver.simpleserver:run_time' \
'-H:ReachabilityMetadataResources@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json+api=META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json' \
-H:DriverWarningsCount=3
]
Warning: Option 'FallbackThreshold' is deprecated and might be removed in a future release: It no longer has any effect, and no replacement is available. Please refer to the GraalVM release notes.
========================================================================================================================
GraalVM Native Image: Generating 'org.apache.avro.avro-ipc_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                   (7.0s @ 0.19GiB)
 Builder configuration:
 - Java version: 25.0.2+10-LTS, vendor version: Oracle GraalVM 25.1.0-dev+10.1
 - Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 - C compiler: gcc (linux, x86_64, 13.3.0)
 - Assertions: disabled, system assertions: disabled
 - 1 user-specific feature(s):
   - org.graalvm.junit.platform.JUnitPlatformFeature
 Image configuration:
 - Garbage collector: Serial GC (max heap size: 80% of RAM)
 - Assertions: disabled (class-specific config may apply), system assertions: disabled
------------------------------------------------------------------------------------------------------------------------
 1 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 13.27GiB of memory (85% of system memory because $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.avro/avro-ipc/1.10.2/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                  (38.6s @ 1.04GiB)
    9,997 types,  11,837 fields, and  46,319 methods found reachable
      510 types,     119 fields, and   1,250 methods registered for reflection
       60 types,      68 fields, and      52 methods registered for JNI access
      157 resource accesses registered with 262.61KiB total size
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                              (5.8s @ 0.99GiB)
[4/8] Parsing methods...      [**]                                                                      (4.1s @ 1.04GiB)
[5/8] Inlining methods...     [***]                                                                     (2.0s @ 1.11GiB)
[6/8] Compiling methods...    [*****]                                                                  (23.2s @ 0.94GiB)
[7/8] Laying out methods...   [**]                                                                      (3.3s @ 1.14GiB)
[8/8] Creating image...       [**]                                                                      (3.7s @ 1.33GiB)
  11.11MiB (34.53%) for code area:    29,413 compilation units
  16.56MiB (51.50%) for image heap:  276,213 objects and 25 resources
   4.49MiB (13.97%) for other data
  32.16MiB in total image size, 29.69MiB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   1.21MiB o.g.n.runtime/com.oracle.svm.core.code              3.49MiB byte[] for code metadata
 747.24KiB java.base/java.util                                 3.14MiB byte[] for string data
 408.46KiB java.base/jdk.internal.classfile.impl               1.65MiB java.base/java.lang.String
 345.35KiB java.base/java.lang                                 1.21MiB java.base/java.lang.Class
 278.72KiB java.base/java.text                                 1.07MiB o.g.n.runtime/c.o.s.core.hub.DynamicHubCompanion
 267.01KiB java.base/com.sun.crypto.provider                 942.01KiB byte[] for general heap data
 246.63KiB java.base/java.util.concurrent                    558.20KiB byte[] for reflection metadata
 230.88KiB com.fasterxml.jackson.databind.deser.std          386.84KiB o.g.n.runtime/c.o.s.c.h.ImageReflectionMetadata
 223.43KiB org.assertj.core.internal.bytebuddy.jar.asm       368.58KiB java.base/java.lang.Object[]
 192.89KiB java.base/sun.security.provider                   323.78KiB java.base/j.u.concurrent.ConcurrentHashMap$Node
   7.02MiB for 370 more packages                               3.48MiB for 2360 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 23 component(s) is embedded in binary (1.21KiB). 9 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                       7.3s (8.0% of total time) in 936 GCs | Peak RSS: 2.00GiB | CPU load: 3.64
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.avro/avro-ipc/1.10.2/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.avro/avro-ipc/1.10.2/build/native/nativeTestCompile/org.apache.avro.avro-ipc_tests-tests (executable, 29.69MiB)
 /tmp/native-image-build-output.json (build_info, 1.03KiB)
========================================================================================================================
Finished generating 'org.apache.avro.avro-ipc_tests-tests' in 1m 30s.

The build process encountered 4 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.avro/avro-ipc/1.10.2/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.avro/avro-ipc/1.10.2/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.avro/avro-ipc/1.10.2/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

org_apache_avro.avro_ipc.ReflectRequestorTest > getClientWithExistingReflectRequestorCreatesProxyUsingRequestorClassLoader() FAILED

org_apache_avro.avro_ipc.ReflectRequestorTest > getClientWithReflectDataCreatesProxyThatReportsProtocolAndRemoteName() FAILED

org_apache_avro.avro_ipc.SpecificRequestorTest > getClientCreatesProxyThatReportsProtocolAndRemoteName() SUCCESSFUL

org_apache_avro.avro_ipc.SpecificRequestorTest > getClientWithExistingRequestorCreatesProxyUsingRequestorClassLoader() SUCCESSFUL

org_apache_avro.avro_ipc.SpecificResponderTest > respondInvokesMatchingImplementationMethod() SUCCESSFUL


Failures (2):
  JUnit Jupiter:ReflectRequestorTest:getClientWithExistingReflectRequestorCreatesProxyUsingRequestorClassLoader()
    MethodSource [className = 'org_apache_avro.avro_ipc.ReflectRequestorTest', methodName = 'getClientWithExistingReflectRequestorCreatesProxyUsingRequestorClassLoader', methodParameterTypes = '']
    => java.lang.ExceptionInInitializerError
       org.apache.avro.reflect.ReflectData.getProtocol(ReflectData.java:903)
       org.apache.avro.ipc.reflect.ReflectRequestor.<init>(ReflectRequestor.java:46)
       org_apache_avro.avro_ipc.ReflectRequestorTest.getClientWithExistingReflectRequestorCreatesProxyUsingRequestorClassLoader(ReflectRequestorTest.java:40)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
     Caused by: org.apache.avro.AvroRuntimeException: Unable to load a functional FieldAccess class!
       org.apache.avro.reflect.ReflectionUtil.resetFieldAccess(ReflectionUtil.java:64)
       org.apache.avro.reflect.ReflectionUtil.<clinit>(ReflectionUtil.java:51)
       [...]
  JUnit Jupiter:ReflectRequestorTest:getClientWithReflectDataCreatesProxyThatReportsProtocolAndRemoteName()
    MethodSource [className = 'org_apache_avro.avro_ipc.ReflectRequestorTest', methodName = 'getClientWithReflectDataCreatesProxyThatReportsProtocolAndRemoteName', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class org.apache.avro.reflect.ReflectionUtil
       org.apache.avro.reflect.ReflectData.getProtocol(ReflectData.java:903)
       org.apache.avro.ipc.reflect.ReflectRequestor.getClient(ReflectRequestor.java:75)
       org_apache_avro.avro_ipc.ReflectRequestorTest.getClientWithReflectDataCreatesProxyThatReportsProtocolAndRemoteName(ReflectRequestorTest.java:26)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
       java.base@25.0.2/java.util.concurrent.FutureTask.run(FutureTask.java:328)
       java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
       [...]

Test run finished after 7 ms
[         5 containers found      ]
[         0 containers skipped    ]
[         5 containers started    ]
[         0 containers aborted    ]
[         5 containers successful ]
[         0 containers failed     ]
[         5 tests found           ]
[         0 tests skipped         ]
[         5 tests started         ]
[         0 tests aborted         ]
[         3 tests successful      ]
[         2 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.avro/avro-ipc/1.10.2/build/reports/problems/problems-report.html
SLF4J(W): No SLF4J providers were found.
SLF4J(W): Defaulting to no-operation (NOP) logger implementation
SLF4J(W): See https://www.slf4j.org/codes.html#noProviders for further details.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.avro/avro-ipc/1.10.2/build/native/nativeTestCompile/org.apache.avro.avro-ipc_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1m 33s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
18 actionable tasks: 8 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.apache.avro:avro-ipc:1.10.2 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 35s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 1.12.0 at 2026-05-09T07:11:17Z with exit code 1 after 95s
❌ ERROR: 'native-image run' for 1.12.0 failed with exit code 1.
FAILED[native-image run][1.12.0][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="1.12.0" ./gradlew clean test -Pcoordinates="org.apache.avro:avro-ipc:1.10.2"]
```

--- a/metadata/org.apache.avro/avro-ipc/1.12.0/reachability-metadata.json
+++ b/metadata/org.apache.avro/avro-ipc/1.12.0/reachability-metadata.json
@@ -1,0 +1,160 @@
+{
+  "reflection" : [
+    {
+      "condition" : {
+        "typeReached" : "org.apache.avro.ipc.Requestor"
+      },
+      "type" : "org.apache.avro.ipc.HandshakeRequest",
+      "fields" : [
+        {
+          "name" : "MODEL$"
+        },
+        {
+          "name" : "SCHEMA$"
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.apache.avro.ipc.Responder"
+      },
+      "type" : "org.apache.avro.ipc.HandshakeRequest",
+      "fields" : [
+        {
+          "name" : "SCHEMA$"
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.apache.avro.ipc.Requestor"
+      },
+      "type" : "org.apache.avro.ipc.HandshakeResponse",
+      "fields" : [
+        {
+          "name" : "MODEL$"
+        },
+        {
+          "name" : "SCHEMA$"
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.apache.avro.ipc.Responder"
+      },
+      "type" : "org.apache.avro.ipc.HandshakeResponse",
+      "fields" : [
+        {
+          "name" : "SCHEMA$"
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.apache.avro.ipc.reflect.ReflectRequestor"
+      },
+      "type" : "org.apache.avro.reflect.ReflectionUtil$AccessorTestClass",
+      "fields" : [
+        {
+          "name" : "b"
+        },
+        {
+          "name" : "by"
+        },
+        {
+          "name" : "c"
+        },
+        {
+          "name" : "d"
+        },
+        {
+          "name" : "f"
+        },
+        {
+          "name" : "i"
+        },
+        {
+          "name" : "i2"
+        },
+        {
+          "name" : "l"
+        },
+        {
+          "name" : "o"
+        },
+        {
+          "name" : "s"
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.apache.avro.ipc.HandshakeRequest"
+      },
+      "type" : "org.apache.avro.util.springframework.ConcurrentReferenceHashMap$Segment[]"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.apache.avro.ipc.HandshakeResponse"
+      },
+      "type" : "org.apache.avro.util.springframework.ConcurrentReferenceHashMap$Segment[]"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.apache.avro.ipc.specific.SpecificRequestor"
+      },
+      "type" : "org.apache.avro.util.springframework.ConcurrentReferenceHashMap$Segment[]"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.apache.avro.ipc.specific.SpecificResponder"
+      },
+      "type" : "org.apache.avro.util.springframework.ConcurrentReferenceHashMap$Segment[]"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.apache.avro.ipc.Responder"
+      },
+      "type" : "sun.security.provider.MD5",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    }
+  ],
+  "resources" : [
+    {
+      "condition" : {
+        "typeReached" : "org.apache.avro.ipc.HandshakeRequest"
+      },
+      "glob" : "META-INF/services/org.apache.avro.Conversion"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.apache.avro.ipc.HandshakeResponse"
+      },
+      "glob" : "META-INF/services/org.apache.avro.Conversion"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.apache.avro.ipc.specific.SpecificRequestor"
+      },
+      "glob" : "META-INF/services/org.apache.avro.Conversion"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.apache.avro.ipc.specific.SpecificResponder"
+      },
+      "glob" : "META-INF/services/org.apache.avro.Conversion"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.apache.avro.ipc.HandshakeRequest"
+      },
+      "glob" : "META-INF/services/org.apache.avro.LogicalTypes$LogicalTypeFactory"
+    }
+  ]
+}

--- a/metadata/org.apache.avro/avro-ipc/index.json
+++ b/metadata/org.apache.avro/avro-ipc/index.json
@@ -1,6 +1,24 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "1.12.0",
+    "test-version" : "1.10.2",
+    "source-code-url" : "https://repo.maven.apache.org/maven2/org/apache/avro/avro-ipc/$version$/avro-ipc-$version$-sources.jar",
+    "repository-url" : "https://github.com/apache/avro",
+    "test-code-url" : "https://repo.maven.apache.org/maven2/org/apache/avro/avro-ipc/$version$/avro-ipc-$version$-tests.jar",
+    "documentation-url" : "https://repo.maven.apache.org/maven2/org/apache/avro/avro-ipc/$version$/avro-ipc-$version$-javadoc.jar",
+    "description" : "Apache Avro IPC provides Java inter-process communication and RPC components for Avro protocols. It includes client and server transports, request and response handling, servlet support, and plugins for building Avro-based remote services.",
+    "tested-versions" : [
+      "1.12.0"
+    ],
+    "allowed-packages" : [
+      "org.apache.avro.ipc",
+      "org.apache.avro",
+      "org.apache.avro.reflect",
+      "org.apache.avro.specific"
+    ]
+  },
+  {
     "metadata-version" : "1.10.2",
     "source-code-url" : "https://repo.maven.apache.org/maven2/org/apache/avro/avro-ipc/$version$/avro-ipc-$version$-sources.jar",
     "repository-url" : "https://github.com/apache/avro",

--- a/stats/org.apache.avro/avro-ipc/1.12.0/stats.json
+++ b/stats/org.apache.avro/avro-ipc/1.12.0/stats.json
@@ -1,0 +1,37 @@
+{
+  "versions" : [ {
+    "version" : "1.12.0",
+    "dynamicAccess" : {
+      "breakdown" : {
+        "reflection" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 7,
+          "totalCalls" : 7
+        }
+      },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 7,
+      "totalCalls" : 7
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 378,
+        "missed" : 5653,
+        "ratio" : 0.062676,
+        "total" : 6031
+      },
+      "line" : {
+        "covered" : 84,
+        "missed" : 1319,
+        "ratio" : 0.059872,
+        "total" : 1403
+      },
+      "method" : {
+        "covered" : 23,
+        "missed" : 299,
+        "ratio" : 0.071429,
+        "total" : 322
+      }
+    }
+  } ]
+}

--- a/tests/src/org.apache.avro/avro-ipc/1.10.2/src/test/resources/META-INF/native-image/reachability-metadata.json
+++ b/tests/src/org.apache.avro/avro-ipc/1.10.2/src/test/resources/META-INF/native-image/reachability-metadata.json
@@ -31,6 +31,99 @@
           "org_apache_avro.avro_ipc.SpecificRequestorTest$AvroIpcGreetingService"
         ]
       }
+    },
+    {
+      "condition" : {
+        "typeReached" : "org_apache_avro.avro_ipc.ReflectRequestorTest"
+      },
+      "type" : "com.fasterxml.jackson.databind.ext.Java7SupportImpl",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org_apache_avro.avro_ipc.ReflectRequestorTest"
+      },
+      "type" : "org.apache.avro.util.springframework.ConcurrentReferenceHashMap$Segment[]"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org_apache_avro.avro_ipc.SpecificRequestorTest"
+      },
+      "type" : "org.apache.avro.util.springframework.ConcurrentReferenceHashMap$Segment[]"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.apache.avro.ipc.reflect.ReflectRequestor"
+      },
+      "type" : "org_apache_avro.avro_ipc.ReflectRequestorTest$AvroIpcReflectGreetingService"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.apache.avro.ipc.specific.SpecificRequestor"
+      },
+      "type" : "org_apache_avro.avro_ipc.SpecificRequestorTest$AvroIpcGreetingService",
+      "fields" : [
+        {
+          "name" : "PROTOCOL"
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.apache.avro.ipc.specific.SpecificResponder"
+      },
+      "type" : "org_apache_avro.avro_ipc.SpecificResponderTest$GreetingEndpoint",
+      "methods" : [
+        {
+          "name" : "greet",
+          "parameterTypes" : [
+            "java.lang.CharSequence"
+          ]
+        }
+      ]
+    }
+  ],
+  "resources" : [
+    {
+      "condition" : {
+        "typeReached" : "org_apache_avro.avro_ipc.ReflectRequestorTest"
+      },
+      "glob" : "META-INF/services/org.apache.avro.Conversion"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org_apache_avro.avro_ipc.SpecificRequestorTest"
+      },
+      "glob" : "META-INF/services/org.apache.avro.Conversion"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org_apache_avro.avro_ipc.ReflectRequestorTest"
+      },
+      "glob" : "META-INF/services/org.assertj.core.configuration.Configuration"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org_apache_avro.avro_ipc.ReflectRequestorTest"
+      },
+      "glob" : "META-INF/services/org.assertj.core.presentation.Representation"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org_apache_avro.avro_ipc.ReflectRequestorTest"
+      },
+      "glob" : "META-INF/services/org.slf4j.spi.SLF4JServiceProvider"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org_apache_avro.avro_ipc.ReflectRequestorTest"
+      },
+      "glob" : "org/slf4j/impl/StaticLoggerBinder.class"
     }
   ]
 }

